src/Spice86.Core/Emulator/OperatingSystem/DosFileManager.cs

```diff
@@ -717,4 +762,43 @@ public class DosFileManager {
         dosDiskTransferArea.FileSize = (ushort)attributes.Length;
         dosDiskTransferArea.FileName = Path.GetFileName(matchingFile);
     }
+
+    /// <summary>
+    /// Creates a directory on disk.
+    /// </summary>
+    /// <param name="directory">The directory name to create</param>
+    /// <returns></returns>
+    /// <returns>A <see cref="DosFileOperationResult"/> with details about the result of the operation.</returns>
+    public DosFileOperationResult CreateDirectory(string directory) {
+        string caseInsensitivePath = ToHostFilePath(directory);
+
+        string? fullPath = ToHostCaseSensitiveFileName(directory, false);
+        if (string.IsNullOrWhiteSpace(fullPath)) {
+            return FileNotFoundError(directory);
+        }
+
+        DirectoryInfo? parentFolder = Directory.GetParent(fullPath);
+
+        if (parentFolder?.Exists is false or null) {
+            return FileNotFoundError(directory);
+        }
+
+        if (parentFolder.GetDirectories().Select(x => x.Name)
```

Comment:
As far as I understand, this LINQ expression lists the name of the files and folders of the parent. Maybe it would be good to move it to a method with a name that reflects that (because I had to think about what it does for more that an few seconds :) )

Reply:
Moved to another method.

Sorry about that, I'm a LINQ addict.